tests/passes/infer-share-dom-map/dmap.expect

```diff
@@ -0,0 +1,157 @@
+Domination Map {
```

Comment:
Did you mean to print out the Domination Map here?

Reply:
Right now, I have 2 groups of tests for the infer-share pass: the first prints out the domination map to make sure it's correct, the second tests whether the infer-share pass correctly attaches the "state_share" attribute or not. My thinking behind this was that the first is mainly testing whether domination map analysis is correct, and the second is testing whether my infer-share pass is correct (although this pass obviously relies on my domination map being correct)

Reply:
Okay, if you're going to print out the dominationMap then let's make it a little more human-readable than it is right now. For example, to an external user, these numbers don't obviously correspond to anything. Furthermore, we use the [`get_opts`](https://github.com/cucapra/calyx/blob/master/calyx/src/passes/top_down_compile_control.rs#L612) method to create a new option for the `infer-share` pass to dump out this information when the user provides a CLI option.

Reply:
I can' think of something else other than numbers to print to the d_map. 
Printing the numbers makes the most sense to me; but maybe when I print it I can explain that the numbers correspond to the NODE_ID, BEGIN_ID, and END_ID attributes of the control statements (which are visible at the end of the file)? 
Furthermore, I should definitely print the name of the component of the domination map that I am printing. 
In addition, I can print it a bit more nicely so the user can see very easily which nodes are dominated by which? 
If you have a different suggestion for how I should approach this then lmk.